arangod/Transaction/IndexesSnapshot.cpp

```diff
@@ -0,0 +1,64 @@
+////////////////////////////////////////////////////////////////////////////////
+/// DISCLAIMER
+///
+/// Copyright 2014-2022 ArangoDB GmbH, Cologne, Germany
+/// Copyright 2004-2014 triAGENS GmbH, Cologne, Germany
+///
+/// Licensed under the Apache License, Version 2.0 (the "License");
+/// you may not use this file except in compliance with the License.
+/// You may obtain a copy of the License at
+///
+///     http://www.apache.org/licenses/LICENSE-2.0
+///
+/// Unless required by applicable law or agreed to in writing, software
+/// distributed under the License is distributed on an "AS IS" BASIS,
+/// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+/// See the License for the specific language governing permissions and
+/// limitations under the License.
+///
+/// Copyright holder is ArangoDB GmbH, Cologne, Germany
+///
+/// @author Jan Steemann
+////////////////////////////////////////////////////////////////////////////////
+
+#include "IndexesSnapshot.h"
+
+#include "Indexes/Index.h"
+#include "StorageEngine/PhysicalCollection.h"
+
+namespace arangodb {
+
+IndexesSnapshot::IndexesSnapshot(PhysicalCollection& collection)
+    : _locker(collection._indexesLock, collection._indexesLockWriteOwner,
+              __FILE__, __LINE__),
```

Comment:
I don't really like that we are accessing some internals of the collection here. Can we instead add a `acquireIndexLock` function to `PhysicalCollection` that returns a Locker?

Reply:
Ok, changed